### PR TITLE
[4.0] Test data plugin

### PR DESCRIPTION
**Pull request by brianteeman**
There are a lot of issues with this plugin - see #25777 .

Pull requests in smaller batches to make them easier to test

### Summary of Changes
1. All external links changed to https except for feeds.joomla.org which is not https :(
2. Removed all references to the search component and module

### CODE REVIEW

File: plugins/sampledata/testing/testing.php [hunk 1]
```diff
@@ -4065,26 +3999,6 @@ public function onAjaxSampledataApplyStep8()
...
-				),
-				'params'     => array(
-					'width'      => 20,
-					'button_pos' => 'right',
-					'cache'      => 1,
-					'cache_time' => 900,
-					'cachemode'  => 'itemid'
-				),
-			),
 			array(
```
Comment (wilsonge) [inline plugins/sampledata/testing/testing.php:4002]: You've removed the text for articles 39 and 40 too in the language files as they were also for search https://github.com/joomla/joomla-cms/pull/25802/files#diff-1c2c3645f8a19403a80477874e136344L188

Comment (brianteeman) [reply plugins/sampledata/testing/testing.php:4002]: And?

Comment (wilsonge) [reply plugins/sampledata/testing/testing.php:4002]: You haven't deleted them from the articles to install here - only number 38 is being deleted

Comment (wilsonge) [reply plugins/sampledata/testing/testing.php:4002]: Wait this is module 38 not article 38. ignore me 🤦‍♂

**Comment by richard67** [comment]
@brianteeman What happens with the articles belonging to those removed article titles and texts, e.g. `PLG_SAMPLEDATA_TESTING_SAMPLEDATA_CONTENT_ARTICLE_38_TITLE`? As far as I understand testing.php, they still will be created because there is no check for emtpy title. Or will the model catch that when trying to save? If so, will that result in an error?

**Comment by richard67** [comment]
I have tested this item :red_circle: unsuccessfully on 6c377c5688e48dc21f0e32fd9347efe380dcef3f<br /><br />Without this PR, installation of testing sample data completes.

With this PR, it stops at menu creation and issues 2 PHP notices:

PHP Notice:  Undefined offset: 9 in /home/richard/lamp/public_html/joomla-cms-4.0-dev/plugins/sampledata/testing/testing.php on line 3142
PHP Notice:  Undefined offset: 9 in /home/richard/lamp/public_html/joomla-cms-4.0-dev/plugins/sampledata/testing/testing.php on line 3153

It seems there are still child menu items of removed menu items, but I haven't found it yet.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25802">issues.joomla.org/tracker/joomla-cms/25802</a>.</sub>

**Comment by Bakual** [comment]
If you want to remove an item completely, you best add some check in the foreach which continues the loop early if one of the removed articles would be created.
Or you renumber the remaining strings.
Simply removing the strings alone will not work.

**Comment by richard67** [comment]
Same for other items than articles, e.g. menu items, as it seems.

**Comment by richard67** [comment]
Maybe it would be easier not to remove content and menu items related to com_search but to replace conent by some article telling about the replacement of com_content by com_finder and where to find the external repository for those who still need it, similar to the notice recently at the developer pages.

**Comment by brianteeman** [comment]
Closing - see comments on #25777

**Pull request closed without merging**